### PR TITLE
Use an iterator over OsString for uumain()

### PR DESCRIPTION
**Pull request by Arcterus**
This is the corresponding PR for uutils/uucore#12.  I believe everything has been updated, so there should be no issues with tests or whatnot.  If there are issues, something has been updated incorrectly.

I am not super happy that I needed to allocate a `Vec` in `src/bin/coreutils.rs` for the generics, but the only other easy solution I could see was using trait objects.

### CODE REVIEW

File: src/uu/uname/src/uname.rs [hunk 1]
```diff
@@ -94,34 +94,43 @@ pub fn uumain(args: Vec<String>) -> i32 {
...
             .help("print the operating system name."))
-        .get_matches_from(&args);
+        .get_matches_from(args);
 
-    let argc = args.len();
     let uname = return_if_err!(1, PlatformInfo::new());
     let mut output = String::new();
 
-    if matches.is_present(OPT_KERNELNAME) || matches.is_present(OPT_ALL) || argc == 1 {
+    let all = matches.is_present(OPT_ALL);
```
Comment (sylvestre) [inline src/uu/uname/src/uname.rs:102]: As you are doing more than just using an iterator, could you please also document in the commit message about that kind of change

Comment (Arcterus) [reply src/uu/uname/src/uname.rs:102]: This is basically to get the iterator to work, as getting the length of the iterator doesn’t work (without adding `Clone` to the iterator in `uucore`).

I can add a comment though.

**Comment by Arcterus** [comment]
It looks like it’s pulling the version of `uucore` that’s on crates.io.

**Comment by Arcterus** [comment]
Everything seems good now.

**Pull request merged**